types/convict/index.d.ts

```diff
@@ -37,6 +37,20 @@ declare namespace convict {
         parse: (content: string) => any;
     }
 
+    type PredefinedFormat =
+        | '*'
+        | 'int'
+        | 'port'
+        | 'windows_named_pipe'
+        | 'port_or_windows_named_pipe'
+        | 'url'
+        | 'email'
+        | 'ipaddress'
+        | 'duration'
+        | 'timestamp'
+        | 'nat'
+        | String;
```

Comment:
Did you mean `string`?

Reply:
`string` cant get autocompletion in vscode, but `String` can (dont know why). Tested in vscode 1.29.1 MacOS.

Reply:
Sorry, I'm not sure what you mean by "get autocompletion".  What behavior is missing/incorrect for `string`?

Reply:
These are used in convict like this:

```typescript
const config = {
  redis: {
    doc: 'Redis URL',
    format: String,
    default: 'redis://127.0.0.1:6379'
  }
}
```

and types cannot be used for the format - if you replace `String` with `string`, you'll see `'string' only refers to a type, but is being used as a value here.ts(2693)`.